data_structures/arrays/median_two_array.py

```diff
@@ -1,18 +1,18 @@
-"""
-https://www.enjoyalgorithms.com/blog/median-of-two-sorted-arrays
-"""
-
-
 def find_median_sorted_arrays(nums1: list[int], nums2: list[int]) -> float:
     """
-    Find the median of two arrays.
+    Find the median of two sorted arrays.
+
+    This implementation uses binary search to achieve an optimal solution in O(log(min(m, n))) time.
 
     Args:
-        nums1: The first array.
-        nums2: The second array.
+        nums1: The first sorted array.
+        nums2: The second sorted array.
```

Comment:
We need this function to fail if either array is not sorted and we need one or more tests that prove it does so.